core/src/main/scala/kafka/utils/Json.scala

```diff
@@ -59,4 +59,12 @@ object Json {
     }
   }
 
+  /**
+    * Parse a JSON string into either a generic type T, or a Throwable in the case of exception.
+    */
+  def parseTo[T](input: String, klass: Class[T]): Either[Throwable, T] = {
```

Comment:
The idiomatic way of doing this is:

```scala
def parseTo[T](input: String)(implicit tag: ClassTag[T]): Either[Throwable, T] = {
```

Then the caller can just invoke `parseTo[PartitionAssignment](jsonData)`.

Also, should we call it `parseAs` perhaps?